dsgrid/query/derived_dataset.py

```diff
@@ -0,0 +1,283 @@
+from dsgrid.config.dimensions import DimensionModel
+import logging
+from pathlib import Path
+
+import json5
+
+from dsgrid.config.dataset_config import (
+    DataClassificationType,
+    DataSchemaType,
+    InputDatasetType,
+)
+from dsgrid.data_models import serialize_model
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import DSGInvalidDataset
+from dsgrid.query.models import ProjectQueryModel, DatasetMetadataModel, ColumnType
+from dsgrid.query.query_submitter import QuerySubmitterBase
+from dsgrid.registry.dataset_registry import DatasetRegistry
+from dsgrid.registry.dimension_registry import DimensionRegistry
+from dsgrid.registry.registry_manager import RegistryManager
+from dsgrid.utils.files import dump_data
+from dsgrid.utils.spark import read_dataframe, get_unique_values
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_derived_dataset_config_from_query(
+    query_path: Path, dst_path: Path, registry_manager: RegistryManager
+):
+    """Create a DatasetConfigModel and dimensions from a query result.
+
+    Parameters
+    ----------
+    query_path : Path
+        Output directory from a query.
+    dst_path : Path
+        Directory in which to create the dataset config files.
+    registry_manager : RegistryManager
+
+    Returns
+    -------
+    bool
+        Returns True if the operation is successful.
+    """
+    metadata_file = QuerySubmitterBase.metadata_filename(query_path)
+    query_file = QuerySubmitterBase.query_filename(query_path)
+    table_file = QuerySubmitterBase.table_filename(query_path)
+    if not metadata_file.exists() or not query_file.exists() or not table_file.exists():
+        logger.error("%s is not a valid query result directory", query_path)
+        return False
+
+    query = ProjectQueryModel.from_file(query_file)
+    if not does_query_support_a_derived_dataset(query):
+        return False
+
+    metadata = DatasetMetadataModel.from_file(metadata_file)
+    project = registry_manager.project_manager.load_project(
+        query.project.project_id, version=query.project.version
+    )
+    new_supplemental_dims_path = dst_path / "new_supplemental_dimensions"
+    df = read_dataframe(table_file)
+    # TODO: should there be a warning if the current project version is later?
+    dimensions = []
```

Comment:
This is quite an operation here, would help if there's a sentence describing this section